src/battle_script_commands.c

```diff
@@ -16345,3 +16393,31 @@ static void TryUpdateRoundTurnOrder(void)
     }
 }
 
+u8 GetFirstFaintedPartyIndex(u8 battlerId)
+{
+    u32 i;
+    u32 start = 0;
+    u32 end = PARTY_SIZE;
+    struct Pokemon *party = (GetBattlerSide(battlerId) == B_SIDE_PLAYER) ? gPlayerParty : gEnemyParty;
+
+    // Check whether partner is separate trainer.
+    if ((GetBattlerSide(battlerId) == B_SIDE_PLAYER && gBattleTypeFlags & BATTLE_TYPE_INGAME_PARTNER)
+     || (GetBattlerSide(battlerId) == B_SIDE_OPPONENT && gBattleTypeFlags & BATTLE_TYPE_TWO_OPPONENTS))
+    {
+        if (GetBattlerPosition(battlerId) == B_POSITION_OPPONENT_LEFT
+         || GetBattlerPosition(battlerId) == B_POSITION_PLAYER_LEFT)
+            end = PARTY_SIZE / 2;
+        else
+            start = PARTY_SIZE / 2;
+    }
+
+    // Loop through to find fainted battler.
+    for (i = start; i < end; ++i)
+        if (GetMonData(&party[i], MON_DATA_SPECIES_OR_EGG) != SPECIES_NONE
+         && GetMonData(&party[i], MON_DATA_SPECIES_OR_EGG) != SPECIES_EGG
+         && GetMonData(&party[i], MON_DATA_HP) == 0)
+            return i;
```

Comment:
It's very minor but it's more efficient to only call `GetMonData` once where possible (it saves a decrypt and encrypt!)
```suggestion
    for (i = start; i < end; ++i)
    {
        u32 species = GetMonData(&party[i], MON_DATA_SPECIES_OR_EGG);
        if (species != SPECIES_NONE
         && species != SPECIES_EGG
         && GetMonData(&party[i], MON_DATA_HP) == 0)
            return i;
    }
```
I'd probably also put braces around that `if` because it's multi-line (albeit only in the condition), but I'll leave that to your discretion.